lib/managers/files.js

```diff
@@ -1344,6 +1375,53 @@ Files.prototype.getRepresentationInfo = function(fileID, representationType, cal
 		})
 		.asCallback(callback);
 };
+
+/**
+ * Get the contents of a representation of a file, e.g, the binary content of an image or pdf.
+ *
+ * API Endpoint: '/files/:fileID?fields=representations'
+ * Method : GET
+ *
+ * @param {string} fileID The file ID to get the representation of
+ * @param {string} representationType The X-Rep-Hints type to request
+ * @param {Object} [options] Optional parameters
+ * @param {string} [options.assetPath] Asset path for representations with multiple files
+ * @param {Function} [callback] Passed a stream over the representation contents if successful
+ * @returns {Promise<Readable>} A promise resolving to a stream over the representation contents
+ */
+Files.prototype.getRepresentationContent = function(fileID, representationType, options, callback) {
+
+	options = Object.assign({ assetPath: '' }, options);
+
+	return this.getRepresentationInfo(fileID, representationType)
+		.then(reps => {
+
+			var repInfo = reps.entries.pop();
+			if (!repInfo) {
+				throw new Error('Could not get information for requested representation');
+			}
+
+			switch (repInfo.status.state) {
```

Comment:
Note, if `representationType` is the empty string (or indistinguishable from it to the server), the returned object won't have a `status` field and this will (may) fail?

Reply:
That's a good point — I'll add basic sanity checking to ensure that a non-empty string did get passed.